spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/api/ChannelAwareMessageListener.java

```diff
@@ -27,7 +28,7 @@
  * @author Gary Russell
  */
 @FunctionalInterface
-public interface ChannelAwareMessageListener {
+public interface ChannelAwareMessageListener extends MessageListener {
```

Comment:
Having this change, I guess we should reconsider logic in the `AbstractAdaptableMessageListener`.

The `MessageListenerAdapter` has also redundant check now as well:
```
else if (!(delegate instanceof MessageListener)) {
					throw new AmqpIllegalStateException("MessageListenerAdapter cannot handle a "
							+ "ChannelAwareMessageListener delegate if it hasn't been invoked with a Channel itself");
}
```
which is not true after `if (delegate instanceof ChannelAwareMessageListener)` any more.